sdk/tables/Azure.Data.Tables/src/TableBearerTokenChallengeAuthorizationPolicy.cs

```diff
@@ -0,0 +1,106 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Azure.Core;
+using Azure.Core.Pipeline;
+
+namespace Azure.Data.Tables
+{
+    /// <summary>
+    /// The storage authorization policy which supports challenges including tenantId discovery.
+    /// </summary>
+    internal class TableBearerTokenChallengeAuthorizationPolicy : BearerTokenAuthenticationPolicy
```

Comment:
Should we move this into Azure.Core shared source rather than duplicating the Storage policy?
https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/storage/Azure.Storage.Common/src/Shared/StorageBearerTokenChallengeAuthorizationPolicy.cs

Reply:
It's not purely a duplicate - but in either case, the Storage APIs evolve independently of the Tables API, so I wouldn't want to lock them together.